nekoyume/Assets/_Scripts/BlockChain/RPCAgent.cs

```diff
@@ -201,28 +208,36 @@ public FungibleAssetValue GetBalance(Address address, Currency currency)
                 serialized.ElementAt(1).ToBigInteger());
         }
 
-        public async Task<FungibleAssetValue> GetBalanceAsync(Address address, Currency currency)
+        public async Task<FungibleAssetValue> GetBalanceAsync(Address addr, Currency currency)
         {
-            if (Game.Game.instance.CachedBalance.TryGetValue(address, out FungibleAssetValue value) &&
-                !value.Equals(default) && Game.Game.instance.CachedAddresses.TryGetValue(address, out bool cached) &&
-                cached)
+            var game = Game.Game.instance;
+            if (game.CachedBalance.TryGetValue(currency, out var cache) &&
+                cache.TryGetValue(addr, out var fav) &&
+                !fav.Equals(default))
             {
                 await Task.CompletedTask;
-                return value;
+                return fav;
             }
+
             // FIXME: `CurrencyExtension.Serialize()` should be changed to `Currency.Serialize()`.
             byte[] raw = await _service.GetBalance(
-                address.ToByteArray(),
+                addr.ToByteArray(),
                 _codec.Encode(CurrencyExtensions.Serialize(currency)),
                 BlockTipHash.ToByteArray()
             );
-            var serialized = (Bencodex.Types.List) _codec.Decode(raw);
+            var serialized = (List)_codec.Decode(raw);
             var balance = FungibleAssetValue.FromRawValue(
                 new Currency(serialized.ElementAt(0)),
                 serialized.ElementAt(1).ToBigInteger());
-            if (address.Equals(Address))
+            if (addr.Equals(Address))
             {
-                Game.Game.instance.CachedBalance[Address] = balance;
+                if (!game.CachedBalance.ContainsKey(currency))
+                {
+                    game.CachedBalance[currency] =
+                        new LruCache<Address, FungibleAssetValue>(2);
+                }
+
+                game.CachedBalance[currency][Address] = balance;
```

Comment:
This caches balance to currency-address, right?
Then, I think `[addr]` is more accurate than `[Address]` since `Address` is a type